media/src/main/java/com/example/media/camera/CameraSnippets.kt

```diff
@@ -0,0 +1,160 @@
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.safeContentPadding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.compose.LifecycleStartEffect
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewModelScope
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.isGranted
+import com.google.accompanist.permissions.rememberPermissionState
+import com.google.accompanist.permissions.shouldShowRationale
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.awaitCancellation
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+// [START android_media_camera_preview_viewmodel]
+class CameraPreviewViewModel(private val appContext: Context) : ViewModel() {
+    private val _surfaceRequest = MutableStateFlow<SurfaceRequest?>(null)
+    val surfaceRequest: StateFlow<SurfaceRequest?> = _surfaceRequest
+
+    private val previewUseCase = Preview.Builder().build().apply {
+        setSurfaceProvider { newSurfaceRequest ->
+            _surfaceRequest.value = newSurfaceRequest
+        }
+    }
+    private val captureUseCase = ImageCapture.Builder().build()
+    private val useCaseGroup = UseCaseGroup.Builder().apply {
+        addUseCase(previewUseCase)
+        addUseCase(captureUseCase)
+    }.build()
+
+    private var runningCameraJob: Job? = null
+
+    fun startCamera(lifecycleOwner: LifecycleOwner) {
+        viewModelScope.launch {
+            runningCameraJob?.cancelAndJoin()
+            val processCameraProvider = ProcessCameraProvider.awaitInstance(appContext)
+            processCameraProvider.bindToLifecycle(
+                lifecycleOwner,
+                CameraSelector.DEFAULT_BACK_CAMERA,
+                useCaseGroup
+            )
+
+            try {
+                awaitCancellation()
+            } finally {
+                processCameraProvider.unbindAll()
+            }
+        }.also { runningCameraJob = it }
+    }
+
+    fun stopCamera() {
+        runningCameraJob?.cancel()
+    }
+
+    fun takePicture() {
+        viewModelScope.launch {
+            val imageProxy = captureUseCase.takePicture()
```

Comment:
Take picture doesn't compile without parameters passed in

Reply:
Can't reproduce on my side. Code works well both with camerax-compose alpha02 and alpha06

Reply:
This is an extension function, so make sure you `import androidx.camera.core.takePicture`.